File: plot/src/main/java/net/fwitz/math/plot/binary/dual/DualNumberFunctionRenderer.java
package net.fwitz.math.plot.binary.dual;

import net.fwitz.math.binary.dual.DualNumber;
import net.fwitz.math.plot.binary.BinaryNumberFunctionPanel;
import net.fwitz.math.plot.binary.BinaryNumberFunctionRenderer;
import net.fwitz.math.plot.renderer.filter.ValuesFilters;

import static net.fwitz.math.binary.dual.DualNumber.dualNumber;

public class DualNumberFunctionRenderer<V> extends BinaryNumberFunctionRenderer<DualNumber, V> {
    public DualNumberFunctionRenderer(BinaryNumberFunctionPanel<DualNumber, V> panel, Class<V> valueType,
                                      int width, int height,
                                      ValuesFilters<V> valuesFilters) {
        super(panel, valueType, width, height, valuesFilters);
    }

    @Override
    protected DualNumber value(double x, double y) {
        return dualNumber(x, y);
    }
}

File: plot/src/main/kotlin/net/fwitz/math/plot/binary/dual/DualNumberFunctionRenderer.kt
package net.fwitz.math.plot.binary.dual

import net.fwitz.math.binary.dual.DualNumber
import net.fwitz.math.plot.binary.BinaryNumberFunctionPanel
import net.fwitz.math.plot.binary.BinaryNumberFunctionRenderer
import net.fwitz.math.plot.renderer.filter.ValuesFilters

class DualNumberFunctionRenderer<V>(
    panel: BinaryNumberFunctionPanel<DualNumber, V>, valueType: Class<V>,
    width: Int, height: Int,
    valuesFilters: ValuesFilters<V>
) : BinaryNumberFunctionRenderer<DualNumber, V>(panel, valueType, width, height, valuesFilters) {
    override fun value(x: Double, y: Double) = DualNumber(x, y)
}